bucket_map/src/bucket.rs

```diff
@@ -890,6 +890,152 @@ impl<'b, T: Clone + Copy + PartialEq + std::fmt::Debug + 'static> Bucket<T> {
 mod tests {
     use {super::*, crate::index_entry::OccupyIfMatches, tempfile::tempdir};
 
+    #[test]
+    fn test_batch_insert_non_duplicates_reusing_file_many_entries() {
+        // 3 variations of reuse
+        for reuse_type in 0..3 {
+            let data_buckets = Vec::default();
+            let v = 12u64;
+            let random = 1;
+            // with random=1, 6 entries is the most that don't collide on a single hash % cap value.
+            for len in 0..7 {
+                log::error!("testing with {len}");
+                // cannot use pubkey [0,0,...] because that matches a zeroed out default file contents.
+                let raw = (0..len)
+                    .map(|l| (Pubkey::from([(l + 1) as u8; 32]), v + (l as u64)))
+                    .collect::<Vec<_>>();
+
+                let mut hashed = Bucket::index_entries(&raw, random);
+                let hashed_raw = hashed.clone();
+
+                let tmpdir = tempdir().unwrap();
+                let paths: Arc<Vec<PathBuf>> = Arc::new(vec![tmpdir.path().to_path_buf()]);
+                assert!(!paths.is_empty());
+                let max_search = 2;
+                let (mut index, file_name) = BucketStorage::<IndexBucket<u64>>::new(
+                    paths.clone(),
+                    1,
+                    std::mem::size_of::<crate::index_entry::IndexEntry<u64>>() as u64,
+                    max_search,
+                    Arc::default(),
+                    Arc::default(),
+                );
+                index.delete_file_on_drop = false;
+                let cap = index.capacity();
+
+                hashed.sort_unstable_by(|a, b| (a.0 % cap).cmp(&(b.0 % cap)).reverse());
+                hashed.windows(2).for_each(|two| {
+                    assert_ne!(two[0].0 % cap, two[1].0 % cap, "{two:?}, cap: {cap}");
+                });
+
+                // file is blank, so nothing matches, so everything returned in `hashed` to retry.
```

Comment:
nit: The comments seem to contradict with the following assert - `hashed.is_empty()`.

Reply:
I don't see an assert of `hashed.is_empty()`. Just `duplicates.is_empty()`. We assert that `hashed` contains everything we tried to insert after calling `batch_insert_non_duplicates_reusing_file`